### ISSUE TITLE
LATERAL keyword not recognized?

### ISSUE THREAD

**Issue by osvenskan** [sqlfluff/sqlfluff#1040]
When linting, SQLFluff doesn't seem to get that `LATERAL` is a valid keyword, although it has done so in the past based on what I read in #832.

## Sample SQL
```sql
SELECT foo
FROM bar
JOIN LATERAL (
        SELECT stuff
        FROM some_other_table
        WHERE
            foo.bar = some_other_table.bar
) ON TRUE
;
```

## Expected Behaviour

`LATERAL` keyword should be accepted

## Observed Behaviour
```
$ sqlfluff lint bin/sql/test.sql
== [bin/sql/test.sql] FAIL
L:   3 | P:  13 | L017 | Function name not immediately followed by bracket.
L:   7 | P:  13 | L026 | Reference 'foo.bar' refers to table/view 'foo' not found
                       | in the FROM clause or found in parent subquery.
L:   7 | P:  13 | L028 | Qualified reference 'foo.bar' found in single table
                       | select which is inconsistent with previous references.
```

## Version
```
$ sqlfluff --version
sqlfluff, version 0.5.2

$ python -V
Python 3.8.8
```

## Configuration
Using default 0.5.2. config

**Comment by osvenskan** [comment]
I'm using Postgres, and I have  a config file that sets the dialect to Postgres. Using that config, I get the exact same output. I reported this as occurring under the `ansi` dialect because it seemed easier to report that I was using the default config rather than my custom one. Sorry if I made things more difficult instead!

**Comment by barrywhart** [comment]
No worries! There are so many tiny (or sometimes not so tiny) differences between the various versions of SQL, it can make your head spin sometimes. 🤯
